pages/linux/whiptail.md

```diff
@@ -0,0 +1,27 @@
+# whiptail
+
+> Display text-based dialog boxes from shell scripts.
+
+- Display a simple message:
+
+`whiptail --title "{{title}}" --msgbox "{{message}}" {{height_in_chars}} {{width_in_chars}}`
+
+- Display a boolean choice, returning the result through the exit code:
+
+`whiptail --title "{{title}}" --yesno "{{message}}" {{height_in_chars}} {{width_in_chars}}`
+
+- Customise the text on the yes / no buttons:
+
+`whiptail --title "{{title}}" --yes-button "{{text}}" --no-button "{{text}}" --yesno "{{message}}" {{height_in_chars}} {{width_in_chars}}`
+
+- Display a text input box:
+
+`{{result_variable_name}}="$(whiptail --title "{{title}}" --inputbox "{{message}}" {{height_in_chars}} {{width_in_chars}} {{default_text}} 3>&1 1>&2 2>&3)"`
```

Comment:
Oh wow this is so useful and smart! I love it!

Reply:
I don't acutally know precisely how all the shell redirections work, but it does lol